### DISCUSSION TITLE
i get 404 on /livewire/update using mcamara/laravel-localization

### DISCUSSION THREAD

**Discussion by SamirHartlief** [livewire/livewire#6055] [1. Help]
i get 404 on /livewire/update using mcamara/laravel-localization

default page works but not the translated pages (/es, /fr)

**Comment by dragonfly4** [comment]
You need to add the setUpdateRoute function under the LaravelLocalization::setLocale() middleware.

https://livewire.laravel.com/docs/installation#configuring-livewires-update-endpoint

routes/web.php

```PHP
Route::prefix(LaravelLocalization::setLocale())->middleware('localeSessionRedirect', 'localizationRedirect', 'localeViewPath')->group(function () {
    /** ADD ALL LOCALIZED ROUTES INSIDE THIS GROUP **/

    Livewire::setUpdateRoute(function ($handle) {
        return Route::post('/livewire/update', $handle);
    });
```

Ref package: https://github.com/mcamara/laravel-localization

**Reply by sylainx** [reply]
Thank you 😇

**Reply by ryanbadger** [reply]
lifesaver

**Reply by nicoolas3198** [reply]
`if(dragonfly4.age >= 18 && likesbeer){
 return 'Cold beer'
}else{
 return 'Other cold drink'
}` You saved my life

**Reply by dragonfly4** [reply]
> `if(dragonfly4.age >= 18 && likesbeer){ return 'Cold beer' }else{ return 'Other cold drink' }` You saved my life

Cheers 🍻

**Reply by frizzant** [reply]
This took ages for me to figure out where the issue was even rooting from.

**Comment by Briareos17** [comment]
If you do not need all web middlewares, you can simply register livewire route in `AppServiceProvider`'s `boot()` method:

````
Livewire::setUpdateRoute(function ($handle) {
            return Route::prefix(LaravelLocalization::setLocale())->post('livewire/update', $handle)
                ->middleware('livewire');
        });
````

Here I made separate middleware group in `Http\Kernel` with only needed middewares for Livewire.

**Comment by mercury64** [comment]
Why can this not be a variable in the config/livewire.php like the 'asset_url'?
Then we can simply modify the .env or config for each published app.
Seems like alot of overhead for a route config.

The livewire/update could be env('APP_URL') . 'livewire/update'

And when I get to uploading files, I'm sure that I will have an issue with livewire/upload-files as well.

**Comment by hmshilbaya-laravel** [comment]
Ready to go :) Finally solved

[AppServiceProvider.php](https://github.com/user-attachments/files/24639673/AppServiceProvider.php)
